spec.md

```diff
@@ -0,0 +1,123 @@
+# SBD Spec
+
+This SBD spec defines the protocol for SBD servers and clients to communicate.
+SBD is a simple websocket-based message relay protocol. 
+
+## 1. Websocket Stack
+
+The SBD protocol is built upon websockets.
+
+### 1.1. Websocket Configuration
+
+#### 1.1.1. Message and Frame Size
+
+The maximum SBD message size (including 32 byte header) is 20000 bytes.
+
+SBD clients and servers MAY set the max message size in the websocket library to 20000 to help enforce this.
+
+The maximum frame size MUST be set larger than 20000 so that sbd messages always fit in a single websocket frame.
+
+## 2. Cryptography
+
+### 2.1. TLS
+
+Generally over the WAN SBD servers SHOULD be available over TLS (wss://), and on a LAN without (ws://).
```

Comment:
Why `ws://` on LAN? On a home network probably yes, but in a shared space I would prefer `wss://`.

Reply:
We're not going to get TLS certs on the LAN, because there's no DNS to work with. For holochain the peer data will be end-to-end encrypted anyway, so we're not concerned about security per-se... But essentially it's not really possible to do the TLS on that LAN.

Reply:
> We're not going to get TLS certs on the LAN, because there's no DNS to work with. For holochain the peer data will be end-to-end encrypted anyway, so we're not concerned about security per-se... But essentially it's not really possible to do the TLS on that LAN.

unless with a web-of-trust where peers exchange certificates out-of-band